crates/stateless/src/validation.rs

```diff
@@ -86,11 +87,77 @@ pub enum StatelessValidationError {
         expected: B256,
     },
 
+    /// Error during signer recovery.
+    #[error("signer recovery failed")]
+    SignerRecovery,
+
+    /// Error when signature has non-normalized s value in homestead block.
+    #[error("signature s value not normalized for homestead block")]
+    HomesteadSignatureNotNormalized,
+
     /// Custom error.
     #[error("{0}")]
     Custom(&'static str),
 }
 
+/// Verifies a transaction using its signature and the given public key.
+///
+/// Note: If the signature or the public key is incorrect, then this method
+/// will return an error.
+///
+/// Returns the address derived from the public key.
+fn recover_sender(
+    vk: &VerifyingKey,
```

Comment:
This PR requires VerifyingKey which is a secp256k1/k256 specific type that is not exposed via alloy_consensus AFAICT, so we directly import it here

Reply:
Exactly for that reason I don't believe that is the best place to implement signature verification. It even locks you into using k256 which is undesirable.

Ideally the verification as well as the public key wrapper type should be added to `alloy_consensus/crypto`:
That module already contains `recover_signer_unchecked` (pre Homestead) and `recover_signer` (post Homestead).

Reply:
In my opinion, it should either be implemented in Alloy or left to the user of reth-stateless as it is currently.

Reply:
Yep I agree re alloy

Reply:
how about you we solve this with a trait that also has k256 and secp features that we expect to be in sync with alloy's crypto.

this can even be more advanced like

```rust
trait StatelessSignerRecoverable {

  fn recover_signer<Crypto>(&self, is_homestead: bool)
}
```

where crypto is like

```rust
struct K256;
struct Secp;
```

and then knows how to do it?

Reply:
I think we would still need secp and k256 to expose VerifyingKey -- I can prototype this to be more concrete

Reply:
Went with a similar approach but instead changed the public key type to be bytes and then added feature flags on the function